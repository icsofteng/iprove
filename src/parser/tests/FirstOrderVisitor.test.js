import {parse} from '../'

test("Visitor Test relations 1", ()=> {
  expect(parse("dragon(x)")).toMatchSnapshot()
})

test("Visitor Test forall", ()=> {
  expect(parse("forall x dragon(x)")).toMatchSnapshot()
})

test("Visitor Test exists", ()=> {
  expect(parse("exists x dragon(x)")).toMatchSnapshot()
})

test("Visitor Test params", ()=> {
  expect(parse("human(x, y, z)")).toMatchSnapshot()
})

test("Visitor Test relations for person", ()=> {
  expect(parse("person(Frank)")).toMatchSnapshot()
})

test("Visitor Test forall with type", ()=> {
  expect(parse("forall x:Int dragon(x)")).toMatchSnapshot()
})

test("Visitor Test less than", () => {
  expect(parse("2 < 3")).toMatchSnapshot()
})

test("Visitor Test less than equal", () => {
  expect(parse("2 <= 3")).toMatchSnapshot()
})

test("Visitor Test greater than", () => {
  expect(parse("2 > 3")).toMatchSnapshot()
})

test("Visitor Test greater than equal", () => {
  expect(parse("2 >= 3")).toMatchSnapshot()
})

test("Visitor Test equal", () => {
  expect(parse("2 == 3")).toMatchSnapshot()
})

test("Visitor Test plus", () => {
  expect(parse("2 + 3")).toMatchSnapshot()
})

test("Visitor Test minus", () => {
  expect(parse("2 - 3")).toMatchSnapshot()
})

test("Visitor Test power", () => {
  expect(parse("2 ^ 3")).toMatchSnapshot()
})

test("Visitor Test multiply", () => {
  expect(parse("2 * 3")).toMatchSnapshot()
})

test("Visitor Test divide", () => {
  expect(parse("2 / 3")).toMatchSnapshot()
})
