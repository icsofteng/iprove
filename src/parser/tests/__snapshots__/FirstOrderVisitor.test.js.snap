// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor Test divide 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "divide",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test equal 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "equal",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test exists 1`] = `
Object {
  "ast": Array [
    Object {
      "symbol": "exists",
      "type": "existential_quantifier",
      "value": Object {
        "name": "dragon",
        "params": Array [
          Object {
            "type": "identifier",
            "value": "x",
            "varType": "Bool",
          },
        ],
        "type": "relation",
      },
      "variables": Array [
        Object {
          "type": "identifier",
          "value": "x",
          "varType": "Bool",
        },
      ],
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [
    Object {
      "name": "dragon",
      "params": Array [
        "Bool",
      ],
    },
  ],
  "types": Array [],
}
`;

exports[`Visitor Test forall 1`] = `
Object {
  "ast": Array [
    Object {
      "symbol": "forall",
      "type": "universal_quantifier",
      "value": Object {
        "name": "dragon",
        "params": Array [
          Object {
            "type": "identifier",
            "value": "x",
            "varType": "Bool",
          },
        ],
        "type": "relation",
      },
      "variables": Array [
        Object {
          "type": "identifier",
          "value": "x",
          "varType": "Bool",
        },
      ],
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [
    Object {
      "name": "dragon",
      "params": Array [
        "Bool",
      ],
    },
  ],
  "types": Array [],
}
`;

exports[`Visitor Test forall with type 1`] = `
Object {
  "ast": Array [
    Object {
      "symbol": "forall",
      "type": "universal_quantifier",
      "value": Object {
        "name": "dragon",
        "params": Array [
          Object {
            "type": "identifier",
            "value": "x",
            "varType": "Int",
          },
        ],
        "type": "relation",
      },
      "variables": Array [
        Object {
          "type": "identifier",
          "value": "x",
          "varType": "Int",
        },
      ],
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [
    Object {
      "name": "dragon",
      "params": Array [
        "Int",
      ],
    },
  ],
  "types": Array [],
}
`;

exports[`Visitor Test greater than 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "greater_than",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test greater than equal 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "greater_than_eq",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test less than 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "less_than",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test less than equal 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "less_than_eq",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test minus 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "minus",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test multiply 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "multiply",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test params 1`] = `
Object {
  "ast": Array [
    Object {
      "name": "human",
      "params": Array [
        Object {
          "type": "identifier",
          "value": "x",
          "varType": "Bool",
        },
        Object {
          "type": "identifier",
          "value": "y",
          "varType": "Bool",
        },
        Object {
          "type": "identifier",
          "value": "z",
          "varType": "Bool",
        },
      ],
      "type": "relation",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "z",
      "varType": "Bool",
    },
    Object {
      "value": "y",
      "varType": "Bool",
    },
    Object {
      "value": "x",
      "varType": "Bool",
    },
  ],
  "relations": Array [
    Object {
      "name": "human",
      "params": Array [
        "Bool",
        "Bool",
        "Bool",
      ],
    },
  ],
  "types": Array [],
}
`;

exports[`Visitor Test plus 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "plus",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test power 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "integer",
        "value": 2,
      },
      "rhs": Object {
        "type": "integer",
        "value": 3,
      },
      "symbol": "power",
      "type": "binary_numerical",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test relations 1 1`] = `
Object {
  "ast": Array [
    Object {
      "name": "dragon",
      "params": Array [
        Object {
          "type": "identifier",
          "value": "x",
          "varType": "Bool",
        },
      ],
      "type": "relation",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "x",
      "varType": "Bool",
    },
  ],
  "relations": Array [
    Object {
      "name": "dragon",
      "params": Array [
        "Bool",
      ],
    },
  ],
  "types": Array [],
}
`;

exports[`Visitor Test relations for person 1`] = `
Object {
  "ast": Array [
    Object {
      "name": "person",
      "params": Array [
        Object {
          "type": "identifier",
          "value": "Frank",
          "varType": "Bool",
        },
      ],
      "type": "relation",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "Frank",
      "varType": "Bool",
    },
  ],
  "relations": Array [
    Object {
      "name": "person",
      "params": Array [
        "Bool",
      ],
    },
  ],
  "types": Array [],
}
`;
